jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -275,21 +300,143 @@ ecma_check_value_type_is_spec_defined (ecma_value_t value) /**< ecma value */
 inline ecma_value_t __attr_const___ __attr_always_inline___
 ecma_make_simple_value (const ecma_simple_value_t simple_value) /**< simple value */
 {
-  return (((ecma_value_t) (simple_value)) << ECMA_VALUE_SHIFT) | ECMA_TYPE_SIMPLE;
+  return (((ecma_value_t) (simple_value)) << ECMA_DIRECT_SHIFT) | ECMA_DIRECT_TYPE_SIMPLE_VALUE;
 } /* ecma_make_simple_value */
 
 /**
- * Number value constructor
+ * Encode an integer number into an ecma-value without allocating memory
+ *
+ * Note:
+ *   The value must fit into the range of allowed ecma integer values
+ *
+ * @return ecma-value
  */
-ecma_value_t __attr_const___
-ecma_make_number_value (const ecma_number_t *ecma_num_p) /**< number to reference in value */
+inline ecma_value_t __attr_const___ __attr_always_inline___
+ecma_make_integer_value (ecma_integer_value_t integer_value) /**< integer number to be encoded */
+{
+  JERRY_ASSERT (ECMA_IS_INTEGER_NUMBER (integer_value));
+
+  return (ecma_value_t) ((integer_value << ECMA_DIRECT_SHIFT) | ECMA_TYPE_DIRECT);
+} /* ecma_make_integer_value */
+
+/**
+ * Allocate and initialize a new float number without checks.
+ *
+ * @return ecma-value
+ */
+static ecma_value_t __attr_const___
+ecma_create_float_number (ecma_number_t ecma_number) /**< value of the float number */
+{
+  ecma_number_t *ecma_num_p = ecma_alloc_number ();
+
+  *ecma_num_p = ecma_number;
+
+  return ecma_pointer_to_ecma_value (ecma_num_p) | ECMA_TYPE_FLOAT;
+} /* ecma_create_float_number */
+
+/**
+ * Create a new float number.
+ *
+ * This function is intended to create floating point constants when it is
+ * known that ecma_number is not an integer number, e.g. 0.5, NaN, or -infinity.
+ * Calling this function is faster than ecma_make_number_value since no
+ * checks are performed on the input apart from a debug assert.
+ *
+ * @return ecma-value
+ */
+inline ecma_value_t __attr_always_inline___
+ecma_make_float_number_value (ecma_number_t ecma_number) /**< value of the float number */
+{
+  /* Simple check to ensure that ecma_number cannot be converted to integer. */
+  JERRY_ASSERT (((ecma_number_t) (ecma_integer_value_t) ecma_number) != ecma_number);
+
+  return ecma_create_float_number (ecma_number);
+} /* ecma_make_float_number_value */
+
+/**
+ * Checks whether the passed number is +0.0
+ *
+ * @return true, if it is +0.0, false otherwise
+ */
+static inline bool __attr_const___ __attr_always_inline___
+ecma_is_number_equals_to_positive_zero (ecma_number_t ecma_number) /**< number */
```

Comment:
I've mentioned this at least twice in previous review rounds (not sure why they don't show up anymore). That's why I asked whether it's been intentionally negliged. Properly, this would be `ecma_is_number_equal_to_positive_zero` or `ecma_number_equals_to_positive_zero`.